.gitlab-ci.yml

```diff
@@ -88,86 +112,120 @@ Docker:
     - docker rmi ${DOCKER_IMAGE}
     - docker logout $CI_REGISTRY
 
-# Deploy feature branch (for specific review apps)
+# Deploy feature branch (for custom backend review apps)
+Setup-review:
+  stage: setup
+  image: dockerhub.ebi.ac.uk/ensembl-web/deploy-tools:latest
+  environment:
+    name: development
+  variables:
+    GIT_STRATEGY: none # ToolsAPI repo not needed here
+    AGENT: ${DEV_AGENT}
+    NS: ${CI_COMMIT_REF_SLUG}
+  rules: # Run once (when a feature branch is created)
+    - if: $CI_DEPLOY_FREEZE == null &&
+          ($CI_COMMIT_BRANCH != "main" || $CI_COMMIT_BRANCH != "dev") &&
+          $CI_PIPELINE_SOURCE == "push" &&
+          $CI_COMMIT_BEFORE_SHA == "0000000000000000000000000000000000000000"
+      when: always
+  script:
+    - kubectl config use-context ${AGENT}
+    # When needed: remove existing ingress, add review namespace & envvars
+    - kubectl config set-context --current --namespace=${DEV_NS}
+    - kubectl delete ingress ensembl-web-tools-api-ingress-${NS} || true
+    - kubectl create namespace ${NS} || true
+    - kubectl config set-context --current --namespace=${NS}
+    - kubectl create secret docker-registry ensemblweb-pull-secret --docker-server=${GITLAB_REGISTRY_URL} 
+      --docker-username=${WEB_PULL_DOCKER_USER} --docker-password=${WEB_PULL_DOCKER_SECRET} --docker-email=${WEB_PULL_DOCKER_EMAIL} || true
+    # Setup ToolsAPI resources in review namespace (service, ingress, secret) + proxy config
+    - git clone --depth 1 --branch k8s123-migration https://gitlab.ebi.ac.uk/ensembl-web/ensembl-k8s-manifests.git
+    - cd ensembl-k8s-manifests/ensembl-web-tools-api
+    - kubectl apply -f service.yaml
+    - sed -i "s#HOST#${NS}.review.ensembl.org#" ingress.yaml
+    - kubectl apply -f ingress.yaml
+    - git clone https://config-ro-user:${CONFIG_RO_TOKEN}@gitlab.ebi.ac.uk/ensembl-web/newsite-api-config-setup.git
+    - cd newsite-api-config-setup/dev/hl
+    - kubectl create secret generic ensembl-web-tools-api-secret --from-env-file=tools-api.secret --dry-run=client -o yaml | kubectl apply -f -
+    - kubectl create configmap proxy-configmap --from-env-file=http-proxy.config --dry-run=client -o yaml | kubectl apply -f -
+
 Review:
-  extends: 
+  extends:
     - .deploy
-    - .deploy-branch-rules
+    - .branch-rules-review
   environment:
     name: development
-  variables: 
+  variables:
     AGENT: ${DEV_AGENT}
     NS: ${CI_COMMIT_REF_SLUG}
-  after_script:
-    - kubectl -n ${NS}  set env deployment/ensembl-tools-api-deployment --from configmap/ensembl-tools-api-configmap
 
-# Stop feature branch review app (cleanup resources) 
+# Stop review app (cleanup backend resources)
 Stop_review:
-  extends: .deploy-branch-rules
   stage: deploy
   image: dockerhub.ebi.ac.uk/ensembl-web/deploy-tools:latest
+  rules:
+    - if: '$CI_DEPLOY_FREEZE == null && ($CI_COMMIT_BRANCH != "main" || $CI_COMMIT_BRANCH != "dev")'
```

Comment:
Why does this job care about CI_DEPLOY_FREEZE?

Reply:
True, it doesn't (as long as the review app was deployed).